packages/frontend/src/renderer/components/CreateJoinCommunity/CreateCommunity/CreateCommunity.test.tsx

```diff
@@ -0,0 +1,260 @@
+      },
+      [NectarStoreKeys.Identity]: {
+        ...new identity.State()
+      }
+    })
+
+    renderComponent(
+      <>
+        <CreateCommunity />
+        <CreateUsername />
+      </>,
+      store
+    )
+
+    // Confirm proper modal title is displayed
+    const dictionary = CreateCommunityDictionary()
+    const createCommunityTitle = screen.getByText(dictionary.header)
+    expect(createCommunityTitle).toBeVisible()
+
+    // Enter community address and hit button
+    const createCommunityInput = screen.getByPlaceholderText(dictionary.placeholder)
+    const createCommunityButton = screen.getByText(dictionary.button)
+    userEvent.type(createCommunityInput, 'rockets')
+    userEvent.click(createCommunityButton)
+
+    // Confirm user is being redirected to username registration
+    const createUsernameTitle = await screen.findByText('Register a username')
+    expect(createUsernameTitle).toBeVisible()
+
+    // Close username registration modal
+    const closeButton = await screen.findByTestId('createUsernameModalActions')
+    userEvent.click(closeButton)
+    expect(createCommunityTitle).toBeVisible()
+  })
+
+  it('user tries to create again a remembered community', async () => {
+    const { store } = await prepareStore({
+      [StoreKeys.Socket]: {
+        ...new SocketState(),
+        isConnected: true
+      },
+      [StoreKeys.Modals]: {
+        ...new ModalsInitialState()
+      },
+      [NectarStoreKeys.Communities]: {
+        ...new communities.State()
+      }
+    })
+
+    const factory = await getFactory(store)
+
+    await factory.create<
+    ReturnType<typeof communities.actions.addNewCommunity>['payload']
+    >('Community')
+
+    renderComponent(
+      <>
+        <CreateCommunity />
+        <CreateUsername />
+      </>,
+      store
+    )
+    store.dispatch(modalsActions.openModal({ name: ModalName.createCommunityModal }))
+    const createUsernameTitle = screen.getByText('Register a username')
+    expect(createUsernameTitle).toBeVisible()
+  })
+
+  it('creates community on submit if connection is ready', async () => {
+    const handleCommunityAction = jest.fn()
+    const component = <PerformCommunityActionComponent
+      open={true}
+      handleClose={() => { }}
+      communityAction={CommunityAction.Create}
+      handleCommunityAction={handleCommunityAction}
+      handleRedirection={() => { }}
+      isConnectionReady={true}
+      community={false}
+    />
+    const result = renderComponent(component)
+    const communityName = 'communityname'
+    const textInput = result.queryByPlaceholderText('Community name')
+    expect(textInput).not.toBeNull()
+    userEvent.type(textInput, communityName)
+    const submitButton = result.queryByRole('button')
+    expect(submitButton).not.toBeNull()
+    expect(submitButton).toBeEnabled()
+    userEvent.click(submitButton)
+    await waitFor(() => expect(handleCommunityAction).toBeCalledWith(communityName))
+  })
+
+  it.each([
```

Comment:
Should this set of tests be copy pasted or a single function somewhere?

Reply:
I don't get it. Could you clarify your expectations?